IO/d4est_norms.c

```diff
@@ -246,11 +255,46 @@ d4est_output_norms
     
 }
 
-
-
+void
+d4est_norms_save
+(
+  const char* options_file,
+  const char** u_name,
+  double** u,
+  double** u_compare,
+  d4est_xyz_fcn_t** analytical_solutions,
+  void* analytical_solution_ctx
+)
+{
+  zlog_category_t *c_default = zlog_get_category("d4est_norms");
+  zlog_debug(c_default, "Saving norms...");
+  
+  // First draft using only u_compare
+  int j = -1;
+  while (u_name[++j] != NULL) {
+    int number_of_nodes = sizeof(*u[j]) / sizeof(u[j][0]);
+    double L_2_sq = 0;
+    double L_infty = 0;
+    for (int i = 0; i < number_of_nodes; i++){
+      double error_i = fabs(u[j][i] - u_compare[j][i]);
+      L_2_sq += pow(error_i, 2);
+      if (error_i > L_infty)
+        L_infty = error_i;
+    }
+    double L_2 = sqrt(L_2_sq);
+    double energy_norm_estimator = -1;
+    
+    char *norms_output_category;
+    asprintf(&norms_output_category, "norms_%s", u_name[j]);
```

Comment:
Is this a good way of concatenating strings, or extremely bad practice? 😄

Reply:
Strings in C are disgusting, but I think this is the best one can do. I get the asprintf idea from the book "21st Century C". However once you are done with norms_output_category you must free it with free(norms_output_category). If you don't it's not the worst thing, but it will create a super small memory leak.